iceoryx_examples/complexdata/iox_publisher_complexdata.cpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+
+std::atomic_bool killswitch{false};
+constexpr char APP_NAME[] = "iox-cpp-publisher-complexdata";
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+// push_front (list), push (stack), emplace_back (vector) return a bool - true if the insertion succeeded, false
+// otherwise
+// to keep the example clear, this helper function handles the return values
+void handleInsertionReturnVal(const bool hasSuccess)
+{
+    if (!hasSuccess)
+    {
+        std::cerr << "Failed to insert element." << std::endl;
+        std::exit(EXIT_FAILURE);
+    }
+}
+
+int main()
+{
+    // register sigHandler
+    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    // initialize publisher
+    iox::popo::Publisher<ComplexDataType> publisher({"Group", "Instance", "Topic"});
+
+    uint64_t ct = 0;
+    // run until interrupted by Ctrl-C
+    while (!killswitch)
+    {
+        ++ct;
+        publisher.loan()
+            .and_then([&](auto& sample) {
+                // forward_list<string<10>, 5>
+                handleInsertionReturnVal(sample->stringForwardList.push_front("world"));
+                handleInsertionReturnVal(sample->stringForwardList.push_front("hello"));
+                // list<uint64_t, 10>;
+                handleInsertionReturnVal(sample->integerList.push_front(ct));
+                handleInsertionReturnVal(sample->integerList.push_front(ct * 2));
+                handleInsertionReturnVal(sample->integerList.push_front(ct + 4));
```

Comment:
Would remove one of those inserts, does not test or demonstrate anything new. (i.e. is just noise)